src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -214,32 +215,44 @@ export class EditDataGridPanel extends GridParentComponent {
 
 		// Setup a function for generating a promise to lookup result subsets
 		this.loadDataFunction = (offset: number, count: number): Promise<{}[]> => {
-			return self.dataService.getEditRows(offset, count).then(result => {
-				let gridData = result.subset.map(r => {
-					let dataWithSchema = {};
-					// skip the first column since its a number column
-					for (let i = 1; i < this.dataSet.columnDefinitions.length; i++) {
-						dataWithSchema[this.dataSet.columnDefinitions[i].field] = {
-							displayValue: r.cells[i - 1].displayValue,
-							ariaLabel: escape(r.cells[i - 1].displayValue),
-							isNull: r.cells[i - 1].isNull
-						};
-					}
-					return dataWithSchema;
-				});
-
-				// should add null row?
-				if (offset + count > this.dataSet.totalRows - 1) {
-					gridData.push(this.dataSet.columnDefinitions.reduce((p, c) => {
-						if (c.id !== 'rowNumber') {
-							p[c.field] = { displayValue: 'NULL', ariaLabel: 'NULL', isNull: true };
+			try {
+				// Setup a function for generating a promise to lookup result subsets
+				return self.dataService.getEditRows(offset, count).then(result => {
+					try {
+						let gridData = result.subset.map(r => {
+							let dataWithSchema = {};
+							// skip the first column since its a number column
+							for (let i = 1; i < this.dataSet.columnDefinitions.length; i++) {
+								dataWithSchema[this.dataSet.columnDefinitions[i].field] = {
+									displayValue: r.cells[i - 1].displayValue,
+									ariaLabel: escape(r.cells[i - 1].displayValue),
+									isNull: r.cells[i - 1].isNull
+								};
+							}
+							return dataWithSchema;
+						});
+
+						// should add null row?
+						if (offset + count > this.dataSet.totalRows - 1) {
+							gridData.push(this.dataSet.columnDefinitions.reduce((p, c) => {
+								if (c.id !== 'rowNumber') {
+									p[c.field] = { displayValue: 'NULL', ariaLabel: 'NULL', isNull: true };
+								}
+								return p;
+							}, {}));
 						}
-						return p;
-					}, {}));
-				}
 
-				return gridData;
-			});
+						return gridData;
+					}
+					catch {
+						throw new Error('Failed to load table data');
```

Comment:
why do we catch here if we're just rethrow?

Reply:
This is meant to show a message to the user that the table data failed to load, probably best to raise an error instead.